GAE/src/org/waterforpeople/mapping/app/web/rest/ActionRestService.java

```diff
@@ -360,14 +360,14 @@ private String copyProject(Long targetId, Long folderId) {
         surveysAncestorIds.add(savedProjectCopy.getKey().getId());
 
         for (Survey sourceSurvey : sourceSurveys) {
+            sourceSurvey.setAncestorIds(surveysAncestorIds);
             SurveyDto surveyDto = new SurveyDto();
             surveyDto.setCode(sourceSurvey.getCode());
             surveyDto.setName(sourceSurvey.getName());
             surveyDto.setPath(projectCopy.getPath() + "/" + sourceSurvey.getName());
             surveyDto.setSurveyGroupId(savedProjectCopy.getKey().getId());
             Survey surveyCopy = SurveyUtils.copySurvey(sourceSurvey, surveyDto);
             surveyCopy.setSurveyGroupId(savedProjectCopy.getKey().getId());
-            sourceSurvey.setAncestorIds(surveysAncestorIds);
```

Comment:
Actually, it may make more sense to set the ancestorIds of the `surveyCopy` so as the avoid an unnecessary step of setting the `sourceSurvey` and then copying that across.